plugins/meta/flannel/flannel.go

```diff
@@ -146,12 +146,17 @@ func saveScratchNetConf(containerID, dataDir string, netconf []byte) error {
 	return ioutil.WriteFile(path, netconf, 0600)
 }
 
-func consumeScratchNetConf(containerID, dataDir string) ([]byte, error) {
+func consumeScratchNetConf(containerID, dataDir string) (func(), []byte, error) {
 	path := filepath.Join(dataDir, containerID)
-	// Ignore errors when removing - Per spec safe to continue during DEL
-	defer os.Remove(path)
 
-	return ioutil.ReadFile(path)
+	// cleanup should be called after DEL succeed
+	cleanup := func() {
```

Comment:
if you set this to a no-op function on error then the caller can do `cleanup()` every time; doesn't have to know the coupling between error and cleanup.

Reply:
got it, thanks!